docs/src/IconButton.doc.js

```diff
@@ -1,419 +1,862 @@
+    dangerouslySetSvgPath={{ __path: 'M23 5v14a4 4 0 0 1-4 4H5a4 4 0 0 1-4-4v-5.5h10.258l-1.94 1.939a1.5 1.5 0 0 0 2.121 2.122L17 12l-5.561-5.561a1.501 1.501 0 0 0-2.121 2.122l1.94 1.939H1V5a4 4 0 0 1 4-4h14a4 4 0 0 1 4 4'}}
+  />
+</Tooltip>
+`}
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="Selected state"
+      description={`IconButton has a \`selected\` state to visually indicate that the element is selected, open, and/or active. If the selected state  controls the display of a Popover-based component (open/closed), use \`accessibilityExpanded\` to inform screen reader users. See the [Accessibility](#Keyboard-interaction) guidelines to learn more.`}
+    >
+      <MainSection.Card
+        cardSize="lg"
+        defaultCode={`
+function SectionsIconButtonDropdownExample() {
+  const [open, setOpen] = React.useState(false);
+  const [selected, setSelected] = React.useState([]);
+  const anchorRef = React.useRef(null);
+  const onSelect = ({ item }) => {
+    if (selected.some(({ value }) => value === item.value )) {
+      setSelected((selected) => selected.filter(({ value }) => value != item.value));
+    } else {
+      setSelected((selected) => [...selected, item]);
+    }
+  };
+
+  return (
+    <Flex justifyContent="center">
+      <IconButton
+        accessibilityControls="sections-dropdown-example"
+        accessibilityExpanded={open}
+        accessibilityHaspopup
+        accessibilityLabel="More Options"
+        bgColor="lightGray"
+        icon="add"
+        iconColor="darkGray"
+        onClick={() => setOpen((prevVal) => !prevVal)}
+        ref={anchorRef}
+        selected={open}
+        size="lg"
+      />
+      {open && (
+        <Dropdown anchor={anchorRef.current} id="sections-dropdown-example" onDismiss={() => setOpen(false)}>
+          <Dropdown.Section label="Create">
+            <Dropdown.Item
+              onSelect={onSelect}
+              option={{ value: 'Pin', label: 'Pin' }}
+              selected={selected}
+            />
+            <Dropdown.Item
+              onSelect={onSelect}
+              option={{ value: 'Story Pin', label: 'Story Pin' }}
+              selected={selected}
+            />
+          </Dropdown.Section>
+          <Dropdown.Section label="Add">
+            <Dropdown.Item
+              badgeText="New"
+              onSelect={onSelect}
+              option={{ value: 'Note', label: 'Note' }}
+              selected={selected}
+            />
+            <Dropdown.Item
+              onSelect={onSelect}
+              option={{ value: 'Section', label: 'Section' }}
+              selected={selected}
+            />
+          </Dropdown.Section>
+        </Dropdown>
+      )}
+    </Flex>
+  );
+}
+      `}
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="Ref"
+      description={`To control focus or position and anchor components to IconButton, use \`ref\` as shown in the example below.`}
```

Comment:
To control focus, or position any anchor components to IconButton, use \`ref\`, as shown in the example below.

Reply:
DONE